### ISSUE TITLE
Add @ApiStatus.Internal annotation on non-API code

### ISSUE THREAD

**Issue by fbricon** [redhat-developer/lsp4ij#380]
This is will make it easier to understand what is considered API or not.

**Comment by angelozerr** [comment]
@CppCXY @InSyncWithFoo @seachicken @KUGDev @AlexWeinstein92 @aparnamichael @MituuZ @turkeylurkey @mrglavas @koxudaxi @FalsePattern please follow this issue.

We wanted to do that in 0.0.2, but as you have reported so many issues (thanks for that!) we had no time to specifiy proper API. For the moment the clean API is only https://github.com/redhat-developer/lsp4ij/blob/main/src/main/java/com/redhat/devtools/lsp4ij/LanguageServerManager.java

For the other classes, please try to avoid using it (except perhaps LSP4IJUtils) because we will mark as @ApiStatus.Internal a lot of classes.

If you need to use some classes, please report that in this issue to discuss together and provide the best API. The main goal of this issue is to provide a clean and robust API in 0.0.3 to avoid breaking something in the future.

Hope you will understand that.

**Comment by KUGDev** [comment]
+1 for @InSyncWithFoo 
We are totally ok with the annotation. But I suppose it to mean that the other available parts should be properly documented for the correct usage or modification to avoid the situation where it is impossible to introduce necessary changes without changing the things marked with the annotation. Of course, it is not a mandatory for the non-GA version, from my perspective

**Comment by angelozerr** [comment]
@InSyncWithFoo don't be afraid with the API described in https://github.com/redhat-developer/lsp4ij/blob/main/docs/DeveloperGuide.md it will not change.

The API which could change is for instance when you want to collect language server from a given file with https://github.com/redhat-developer/lsp4ij/blob/c45ed607afe4c1364b20cd8a8884e74874d997a4/src/main/java/com/redhat/devtools/lsp4ij/LanguageServiceAccessor.java#L43

We would like providing low level API instead of providing a lot of methods hosted in LanguageServiceAccessor that we could refactor because this class comes from https://github.com/eclipse/lsp4e/blob/main/org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServiceAccessor.java which was adapted for IJ. I don't master the code of this class and I spend my time to clean up to have a better base code.

**Comment by angelozerr** [comment]
It is this kind of feedback that I love, thanks @InSyncWithFoo !

We need to think about providing this similar API.

**Comment by mrglavas** [comment]
@angelozerr @fbricon This is going to be extremely helpful. Thank you for doing this.

So far we've been borrowing LSP4MP4iJ from Quarkus Tools to support MicroProfile and then applying patterns that we've seen in Quarkus Tools to our other LSP clients in Liberty Tools. Our assumption has been that if you're using it in Quarkus Tools you must have considered to be an API (at least at that moment in time) otherwise you wouldn't have been using it yourselves.

**Comment by angelozerr** [comment]
> Our assumption has been that if you're using it in Quarkus Tools you must have considered it to be an API (at least at that moment in time) otherwise you wouldn't have been using it yourselves.

Indeed, but it is not true today, it was hard to provide the best LSP support and take care of API both.
